### ISSUE TITLE
QMolecule Design 

### ISSUE THREAD

**Issue by pbark** [Qiskit/qiskit-nature#148]
Aligned with the drivers refactoring we need to converge to a new design for the QMolecule

**Comment by mrossinek** [comment]
With the updated description of this issue above, I propose to convert this issue into an Epic after everyone had time to review it. I will create separate implementation-oriented issues for the three mentioned subtasks which can then be assigned to sprints in order to gradually close this (then) Epic.

I would like to stress the benefit of the proposed step-by-step implementation again, because it will allow us to start working on a prototype for the containers while working on #147 in parallel. If we get towards the end of #147 before reaching the third proposed subtask of this issue we will also have gained a better overview of the drivers themselves which will help in the discussion of step 3 once we get to it.

**Comment by mrossinek** [comment]
As part of #173 we discussed the possibility to further split `qiskit_nature.drivers.second_quantization` into:
```
qiskit_nature/
    drivers/
        second_quantization/
            electronic/
                <...>
            vibrational/
                <...>
```
As such, renaming `QMolecule -> ElectronicDriverResult` and `WatsonHamiltonian -> VibrationalDriverResult` would come very naturally. However, we decided to postpone this until we have a clearer picture of how the implementation of this issue will affect the drivers. Thus, this should be revisited as part of **Step 3.** in the original issue description.
